src/modules/cache-client/cache-client.service.ts

```diff
@@ -76,12 +76,24 @@ export class CacheClient implements ICacheClient {
   async authenticate() {
     let tokens: AuthTokens | undefined = undefined;
 
+    const setTokens = () => {
+      if (!tokens) return;
+      if (!this.isBrowser) {
+        this._httpClient.defaults.headers.common[
+          'Authorization'
+        ] = `Bearer ${tokens.token}`;
+      }
+      this.refresh_token = tokens.refreshToken;
+    };
+
     // First try to refresh access token
     try {
       const refreshedTokens = await this.refreshToken();
+      tokens = refreshedTokens;
+      setTokens();
 
-      if (refreshedTokens && (await this.isAuthenticated())) {
-        tokens = refreshedTokens;
+      if (!(await this.isAuthenticated())) {
```

Comment:
We probably should also find out how authentication may fail right after refreshing token

Reply:
it shouldn't but you know, better to check it :smile: